src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -200,17 +204,32 @@ public override object VisitSqlRoot(SqlRootExpression expression, SearchOptions
                 if (!searchOptions.CountOnly)
                 {
                     StringBuilder.Append("ORDER BY ");
-                    if (searchParamInfo == null || searchParamInfo.Code == KnownQueryParameterNames.LastUpdated)
+
+                    if (_rootExpression.SearchParamTableExpressions.Any(t => t.Kind == SearchParamTableExpressionKind.Include))
                     {
-                        StringBuilder
-                            .Append(VLatest.Resource.ResourceSurrogateId, resourceTableAlias).Append(" ")
-                            .AppendLine(sortOrder == SortOrder.Ascending ? "ASC" : "DESC");
+                        // ensure the matches appear before includes
+                        StringBuilder.Append("IsMatch DESC, ");
+                    }
+
+                    if (isSpecialCaseSort)
```

Comment:
>isSpecialCaseSort [](http://example.com/codeflow?start=24&length=17)

if previous if condition is true and this one as well
we will have double comma.

`IsMatch Desc,,`